lib/controllers/responses.js

```diff
@@ -429,53 +388,100 @@ exports.patch = function patch(req, res) {
 
 
 exports.del = function del(req, res) {
-  Response.findOne({
+  Response.findOneAndUpdateAsync({
     'properties.survey': req.params.surveyId,
     'entries._id': req.params.responseId
-  })
-  .exec(function (error, doc) {
-    if (util.handleError(error, res)) { return; }
-    if (doc === null) {
+  }, {
+    $pull: {
```

Comment:
Might be good to have a brief note about `$pull` functionality since it's not a commonly used operator. Maybe "$pull removes the result of the query from the array". Or maybe that's obvious from the name (I had to look it up when you first proposed it)